rest/rest.go

```
@@ -40,91 +44,153 @@ func New(opts ...Option) *Rest {
+					tracer.WithSpanType(tracer.SpanTypeExit),
+					tracer.WithSpanLayer(tracer.SpanLayerHttp),
+				)
+
+				// Injects X-Request-Id to the request by reading from the context
+				requestID, found := ctx.Value(XRequestId).(string)
+				if !found {
+					requestID = fmt.Sprintf(
+						"%s-%s", uuid.New().String(),
+						time.Now().Format(time.RFC3339),
+					)
+					ctx = context.WithValue(ctx, XRequestId, requestID)
```

golangci-lint (1.62.2)

SA1029: should not use built-in type string as key for value; define your own type to avoid collisions

(staticcheck)